src/modules/service/WebService.ts

```diff
@@ -47,22 +62,33 @@ export class WebService {
     /**
      * The instance of SocketIO
      */
-    protected _socket: SocketIO | null = null;
+    protected _socket: SocketIO;
 
     /**
      * Constructor
      * @param port The bind port
+     * @param private_port The bind private port
      * @param address The bind address
      */
-    constructor(port: number | string, address?: string) {
+    constructor(port: number | string, private_port: number | string, address?: string) {
         if (typeof port === "string") this.port = parseInt(port, 10);
         else this.port = port;
 
+        if (typeof private_port === "string") this.private_port = parseInt(private_port, 10);
+        else this.private_port = private_port;
+
         if (address !== undefined) this.address = address;
         else this.address = "";
 
         this.app = express();
+        this.private_app = express();
         this.eventDispatcher = new EventDispatcher();
+
+        // Create HTTP server.
+        this.server = http.createServer(this.app);
+        this.private_server = http.createServer(this.private_app);
+
+        this._socket = new SocketIO(this.server);
```

Comment:
It would be better to be created within the original start().
This is to open the socket after the route and port settings are completed rather than the creator.

Reply:
Yes, I changed the call location.